### DISCUSSION TITLE
fuse mount after rename directory no content found

### DISCUSSION THREAD

**Discussion by summerisc** [seaweedfs/seaweedfs#1959] [Q&A] [answered]
Steps to replicate:

1. copy a folder from the local disk to the mounted directory
2. rename the directory( for example: mv /mount/exampleName /mount/newNameExample)
3. cd newNameExample/ find no data under this folder (Bug is here)
4. check filer data is still under newNameExample/
5. Unmount and remount, the data shows up again under newNameExample/

Any idea why it happens as step 3 describes? Thanks

**Comment by chrislusf** [comment, marked as answer]
could not reproduce. I mounted to `~/tmp/m2`, and have this:

<img width="420" alt="Screen Shot 2021-04-02 at 17 02 15" src="https://user-images.githubusercontent.com/1543151/113462359-52b76580-93d5-11eb-849b-a2ad94845aa8.png">

**Reply by summerisc** [reply]
emm, yeah it's wired that it didn't work on my end. Tried different filer store(redis, MySQL) got the same results, so rule out any filer store break. The workaround for me is just to use cp -a folder1/. folder2/ instead of using mv cmd, that way it will work. I will mark this as resolved but would love to dig deeper if any clues or anything I could check come to your mind. Thank you!

**Reply by chrislusf** [reply]
what is the operating system? Can you use https://github.com/chrislusf/seaweedfs/blob/master/docker/compose/local-dev-compose.yml to reproduce it?

**Reply by summerisc** [reply]
I am able to reproduce with this file. However, seems chrislusf/seaweedfs has no image tagged as local, so I change the tag to latest. my operating system is Ubuntu 18.04.5 LTS, should be a pretty standard one.

**Reply by chrislusf** [reply]
Can you show your steps to reproduce? Here are my steps using the docker compose file.

<img width="1639" alt="Screen Shot 2021-04-03 at 18 37 20" src="https://user-images.githubusercontent.com/1543151/113496163-b44a0380-94ab-11eb-87e2-3f1488299c2c.png">

**Reply by summerisc** [reply]
sure, steps are:
1. using docker-compose local dev to boot up all the services
2. run `sudo ./weed mount -filer=localhost:8888 -dir=/home/usr/fuse -filer.path=/` to mount the point
3. 
![Screen Shot 2021-04-04 at 9 41 13 AM](https://user-images.githubusercontent.com/21230663/113510705-12b5c700-952a-11eb-866c-1a111a610fed.png)

**Reply by chrislusf** [reply]
Here is my output:
<img width="742" alt="Screen Shot 2021-04-04 at 11 28 11" src="https://user-images.githubusercontent.com/1543151/113518085-e786a400-9538-11eb-84b5-53d084f5367a.png">

**Reply by chrislusf** [reply]
My previous output is run by connected into the docker system:
```
~$ docker ps
CONTAINER ID   IMAGE                       COMMAND                  CREATED        STATUS        PORTS                                                                                                            NAMES
c9569f325a80   chrislusf/seaweedfs:local   "/entrypoint.sh -v=4…"   17 hours ago   Up 17 hours   7333/tcp, 8080/tcp, 8333/tcp, 8888/tcp, 9333/tcp, 18080/tcp, 18888/tcp, 19333/tcp                                seaweedfs_mount_1
51965491b7f4   chrislusf/seaweedfs:local   "/entrypoint.sh -v=1…"   17 hours ago   Up 17 hours   7333/tcp, 8080/tcp, 8888/tcp, 9333/tcp, 18080/tcp, 18888/tcp, 19333/tcp, 0.0.0.0:8333->8333/tcp                  seaweedfs_s3_1
61ad4912ca4b   chrislusf/seaweedfs:local   "/entrypoint.sh -v=1…"   17 hours ago   Up 17 hours   7333/tcp, 8080/tcp, 8333/tcp, 9333/tcp, 0.0.0.0:8888->8888/tcp, 18080/tcp, 19333/tcp, 0.0.0.0:18888->18888/tcp   seaweedfs_filer_1
c0e7633f0396   chrislusf/seaweedfs:local   "/entrypoint.sh -v=1…"   17 hours ago   Up 17 hours   7333/tcp, 8333/tcp, 8888/tcp, 0.0.0.0:8080->8080/tcp, 9333/tcp, 18888/tcp, 19333/tcp, 0.0.0.0:18080->18080/tcp   seaweedfs_volume_1
6b7074db605a   chrislusf/seaweedfs:local   "/entrypoint.sh -v=1…"   17 hours ago   Up 17 hours   7333/tcp, 8080/tcp, 8333/tcp, 8888/tcp, 18080/tcp, 0.0.0.0:9333->9333/tcp, 18888/tcp, 0.0.0.0:19333->19333/tcp   seaweedfs_master_1

docker exec -ti seaweedfs_mount_1 /bin/sh
```

Seems you are executing outside of it from your local mount.

**Reply by summerisc** [reply]
By exec inside the container as you did it works fine. but I am just wondering why the local system directory mount is not working as expected? It could be used to mount a local directory right?

**Reply by chrislusf** [reply]
What is your local "weed version"?

**Reply by summerisc** [reply]
I was using version 30GB 2.32 9f00f95 linux amd64, but I just got upgraded to version 30GB 2.36 6b7aa96 linux amd64 and it works right now. So it seems a problem with weed 2.32. Thanks for the help!
